src/gsi/satthin.F90

```diff
@@ -402,7 +403,11 @@ subroutine makegrids(rmesh,ithin,n_tbin)
     itx_all=0
     if(abs(rmesh) <= one .or. ithin <= 0)then
       use_all=.true.
-      itxmax=1e9
+      if (present(itxmax_in)) then
+         itxmax = itxmax_in
```

Comment:
The idea to reduce the itxmax is great.

Might be a little picky, it is better to set 1e7 as a constant in a common place with two reasons below:

- it is more easy to change it to other value in case of there are multiple codes need to use this value
- I would suggest to check if the total number is bigger than this value if applicable in order to avoid potential memory issues.

This is only my two cents. I am not that confidence on it yet because I haven't understand well about the coding style in GSI yet.